crates/iceberg-catalog/src/config.rs

```diff
@@ -147,6 +166,66 @@ where
     duration.num_seconds().to_string().serialize(serializer)
 }
 
+#[derive(Clone, Serialize, Deserialize, PartialEq, veil::Redact)]
+#[serde(rename_all = "snake_case")]
+pub enum OpenFGAAuth {
+    ClientCredentials {
+        client_id: String,
+        #[redact]
+        client_secret: Option<String>,
+    },
+    #[redact(all)]
+    ApiKey(String),
+}
+
+#[derive(Clone, Serialize, Deserialize, PartialEq, veil::Redact)]
+pub struct OpenFGAConfig {
+    /// GRPC Endpoint Url
+    pub endpoint: Url,
+    /// Store Name - if not specified, `lakekeeper` is used.
+    store_name: Option<String>,
+    /// Server Name - Top level requests are made for `server:{server_id`}
+    /// If not specified, 00000000-0000-0000-0000-000000000000 is used.
+    server_id: Option<uuid::Uuid>,
+    /// API-Key. If client-id is specified, this is ignored.
+    pub api_key: Option<String>,
+    /// Client Credentials
+    pub client_id: Option<String>,
+    #[redact]
+    pub client_secret: Option<String>,
```

Comment:
this could become an enum

Reply:
Fixed - I kept the configuration flat though. Let me know what you think